pkg/oam/util/helper.go

```diff
@@ -633,6 +634,82 @@ func RawExtension2AppConfig(raw runtime.RawExtension) (*v1alpha2.ApplicationConf
 	return ac, nil
 }
 
+// RawExtension2Component converts runtime.RawExtention to Component
+func RawExtension2Component(raw runtime.RawExtension) (*v1alpha2.Component, error) {
+	c := &v1alpha2.Component{}
+	b, err := raw.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(b, c); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// AppConfig2ComponentManifests convert AppConfig and Components to a slice of ComponentManifest.
+func AppConfig2ComponentManifests(acRaw runtime.RawExtension, comps []common.RawComponent) ([]*oamtypes.ComponentManifest, error) {
+	var err error
+	ac, err := RawExtension2AppConfig(acRaw)
+	if err != nil {
+		return nil, err
+	}
+	cms := make([]*oamtypes.ComponentManifest, len(ac.Spec.Components))
+	for i, acc := range ac.Spec.Components {
+		cm := &oamtypes.ComponentManifest{
+			Name:         acc.ComponentName,
+			RevisionName: acc.RevisionName,
+		}
+		if acc.ComponentName == "" && acc.RevisionName != "" {
+			// extract comp name from comp revision name
+			splits := strings.Split(acc.RevisionName, "-")
+			cm.Name = strings.Join(splits[0:len(splits)-1], "-")
```

Comment:
why not use the extract func

Reply:
just avoid importing another util lib in an util lib.